### PR TITLE
Stored Bee API

### PR DESCRIPTION
**Pull request by EOT3000**
Adds a way to modify some properties of individual stored entities (bees in beehive), as well as release them individually, without having to release them all.

### CODE REVIEW

**Comment by katekerllenevich** [comment]
I like this, unfortunately your timings is just off. You'll have to rebase this onto 1.20, I'm too tired to review this tonight but I'll take a look at it later.

File: patches/api/0054-Stored-Bee-API.patch [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+diff --git a/src/main/java/org/bukkit/block/EntityBlockStorage.java b/src/main/java/org/bukkit/block/EntityBlockStorage.java
+index 739911cda33b373f99df627a3a378b37d7d461aa..5090d9ecc396474a3a6fb6ef1b6f9fa9a3b40211 100644
+--- a/src/main/java/org/bukkit/block/EntityBlockStorage.java
++++ b/src/main/java/org/bukkit/block/EntityBlockStorage.java
+@@ -3,6 +3,8 @@ package org.bukkit.block;
+ import java.util.List;
+ import org.bukkit.entity.Entity;
+ import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++import org.purpurmc.purpur.entity.StoredEntity;
```
Comment (granny) [inline patches/api/0054-Stored-Bee-API.patch:16]: inline the fqn

File: patches/server/0311-Stored-Bee-API.patch [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+     // Paper start - Add EntityBlockStorage clearEntities
+     public void clearBees() {
+         this.stored.clear();
+@@ -425,9 +461,10 @@ public class BeehiveBlockEntity extends BlockEntity {
+         private BeeReleaseStatus() {}
+     }
+ 
+-    private static class BeeData {
++    // Purpur - change from private to public
++    public static class BeeData {
```
Comment (granny) [inline patches/server/0311-Stored-Bee-API.patch:67]: inline the comment

File: patches/server/0311-Stored-Bee-API.patch [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+         final int minOccupationTicks;
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBeehive.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBeehive.java
+index 380897c010521f368848a3e6986d307cf47ff319..4b8a8296a32b0240d471ddc502ced61d821ce2d3 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBeehive.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBeehive.java
+@@ -5,6 +5,7 @@ import java.util.ArrayList;
+ import java.util.List;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.world.entity.Entity;
++import net.minecraft.world.level.Level;
```
Comment (granny) [inline patches/server/0311-Stored-Bee-API.patch:82]: doesn't look like this is being used anywhere?

File: patches/server/0311-Stored-Bee-API.patch [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+ import net.minecraft.world.level.block.entity.BeehiveBlockEntity.BeeReleaseStatus;
+ import org.bukkit.Location;
+@@ -13,11 +14,29 @@ import org.bukkit.block.Beehive;
+ import org.bukkit.craftbukkit.entity.CraftBee;
+ import org.bukkit.craftbukkit.util.CraftLocation;
+ import org.bukkit.entity.Bee;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++import org.purpurmc.purpur.entity.PurpurStoredBee;
++import org.purpurmc.purpur.entity.StoredEntity;
```
Comment (granny) [inline patches/server/0311-Stored-Bee-API.patch:93]: inline the fqn

**Review by granny — CHANGES_REQUESTED** [review]
Did a quick skim of your PR. For future reference it's also a good idea to create a new branch for every PR you create. Do what Omega said as well.

**Comment by EOT3000** [comment]
I don't understand how github works too well so I'll make another PR with all the requested changes.

**Pull request closed without merging**